src/main/scala/org/constellation/domain/redownload/RedownloadService.scala

```diff
@@ -70,19 +70,33 @@ class RedownloadService[F[_]](
       }
     } yield proposals
 
+  def fetchMissingSnapshots: F[SnapshotsAtHeight] = ??? // TODO: Implement
+
   def checkForAlignmentWithMajoritySnapshot(): F[Unit] =
     for {
       peersProposals <- getPeerProposals()
       createdSnapshots <- getCreatedSnapshots()
+      acceptedSnapshots <- getAcceptedSnapshots()
+
       majorityState = majorityStateChooser.chooseMajorityState(
         createdSnapshots,
         peersProposals
       )
-      // TODO: Calculate diff and call redownload if true
-      _ <- if (shouldRedownload(createdSnapshots, majorityState, snapshotHeightRedownloadDelayInterval)) F.unit
-      else F.unit
+
+      _ <- if (shouldRedownload(acceptedSnapshots, majorityState, snapshotHeightRedownloadDelayInterval)) {
+        val plan = calculateRedownloadPlan(acceptedSnapshots, majorityState)
+        applyRedownloadPlan(plan) // TODO: Actual redownload, rethink splitting the logic
+      } else F.unit
     } yield ()
 
+  private[redownload] def applyRedownloadPlan(plan: RedownloadPlan): F[Unit] = for {
+    missingSnapshots <- fetchMissingSnapshots // TODO: Store these snapshots on disk now?
```

Comment:
Here should be fetch + accept. After the redownload, all snapshots should be stored on disk and all data from snapshot info should be applied (re: `SnapshotService.setSnapshot`)